molecule/core.py

```diff
@@ -164,9 +164,13 @@ def remove_templates(self):
         """
         if os.path.exists(self.config.config['molecule']['rakefile_file']):
             os.remove(self.config.config['molecule']['rakefile_file'])
-        if not self.state.customconf:
-            if os.path.exists(self.config.config['ansible']['config_file']):
-                os.remove(self.config.config['ansible']['config_file'])
+
+        config = self.config.config['ansible']['config_file']
+        if os.path.exists(config):
+            with open(config, 'r') as stream:
+                data = stream.read().splitlines()
+                if '# Molecule managed' in data:
```

Comment:
Maybe nitpicking a tiny bit, but a regex would probably be more efficient.
Something like:
```python
data = stream.read()
if re.search('# Molecule managed', data, re.IGNORECASE):
    os.remove(config)
```